lib/spack/spack/test/versions.py

```diff
@@ -565,3 +565,12 @@ def test_list_highest():
     assert vl2.highest_numeric() is None
     assert vl2.preferred() == Version('develop')
     assert vl2.lowest() == Version('master')
+
+
+def test_verification():
+    with pytest.raises(ValueError):
+        Version("foo 1.2.0")
+    with pytest.raises(ValueError):
+        Version("!")
+    with pytest.raises(ValueError):
+        Version("1!2")
```

Comment:
Minor nitpick for tests:
```suggestion
@pytest.mark.parametrize('version_str', [
    "foo 1.2.0",
    "!",
    "1!2"
])
def test_invalid_versions(version_str):
    """Ensure invalid versions are rejected with a ValueError"""
    with pytest.raises(ValueError):
        Version(version_str)
```
Beware I didn't try the code above, so watch out for typos :slightly_smiling_face:

Reply:
Looks good, I'll add it, if CI complains I'll pull it and fix it.